components/Ethdriver/src/ethdriver.c

```diff
@@ -298,6 +298,22 @@ int client_rx(int *len)
     return ret;
 }
 
+int client_has_rx(void)
+{
+    if (!done_init) {
+        return -1;
+    }
+
+    client_t *client = eth_get_client();
+    assert(client);
+
+    if ((client->pending_rx_head != client->pending_rx_tail) &&
+        (0 == client->should_notify)) {
+        return 1;
```

Comment:
Would it make sense to also return the amount here?

Reply:
For DW's implementation, we just use it as a check that data is available to grab

``` 
    if (ethdriver_has_rx()) {
        ethif_lwip_handle_irq(&_lwip_driver, 0);
    }
```